### DISCUSSION TITLE
memos-proxy 中间服务推送微信公众号信息功能

### DISCUSSION THREAD

**Discussion by ZhaoUncle** [usememos/memos#148] [Ideas]
自己用 python 简单写了个代理服务作为中间转发微信公众号的留言推送到 memos，不过因为 memos api 不能 push 图片，所以我也没整合了。

https://github.com/ZhaoUncle/memos-proxy

**Comment by boojack** [comment]
这块我之前是放在代码里的，但是感觉微信公众号要求太多就给去掉了，弄成中间服务确实很符合使用场景。

不过我看代码里目前好像只支持了文本信息，可以参考我之前的代码，再加个支持语音、图片以及链接分享卡片
https://github.com/boojack/insmemo/blob/master/controllers/wx.ts#L47

**Comment by ZhaoUncle** [comment]
好像是 memos api 没有 post images 的接口，具体我没去细看代码。

**Reply by boojack** [reply]
所有的 api 都可以通过加上 `?openId=xxx` 来调用。例如，上传资源文件（包括图片）：

```
POST /api/resource?openId=xxx
```

**Comment by ZhaoUncle** [comment]
大致明白您的意思了。
从接口集成的角度来看，你的 /api/memo?openId 接口内容除了 content text 之外，应该携带上images 等其他push 字段，要不然还得通过其他接口去集成，反而是内部处理毕竟方便。
当然，自己拿多个接口的数据在合并也行。不过从程序开发的角度来说，应该有程序内部去处理会更简介。其实还是因为我没有推送图片的需求，就没想写了。

**Reply by boojack** [reply]
目前是类似于 markdown，图片只以链接的形式存在 content 里。

BTW，微信的图片消息会直接给出链接地址，直接存为 content 就可以了，不用再上传到 memos。

**Comment by ZhaoUncle** [comment]
我想我明白了，我先尝试下看看。

**Comment by ZhaoUncle** [comment]
api/resource? post 之后不会返回静态地址 path，无法拼接到 url。建议可以加一个返回字段 png 本地路径或者完整uri 返回。我拿到后拼接可以直接放到 api/memo? 会比较方便。比较自己发送的内容，还是希望吧静态资源放到自己的服务器上，而不是markdown 语法使用微信的url 图片地址。
![image](https://user-images.githubusercontent.com/31182637/185826621-1fe9aa32-0fbe-4602-8cd1-da4636fbd9aa.png)

**Comment by ZhaoUncle** [comment]
请忽略，我好像找到你的图片存储规律了，不过还有有返回就更好。

**Comment by ZhaoUncle** [comment]
video 模式，memos 好像还没适配相应的js组件？或者说还不支持相关功能？

**Reply by lqwakeup** [reply]
目前是不支持显示 video 的，不过可以先和图片一样处理，存到 resource 里。

**Comment by lclrc** [comment]
[lclrc/memos-wechat-proxy](https://github.com/lclrc/memos-wechat-proxy)
在原repo的基础上添加了视频/语音/卡片链接的支持, 修复了图片支持, 添加可自定义默认标签

**Comment by athurg** [comment]
微信公众号最大的问题就是：

- 如果你是非认证用户，那么公众号只能被折叠到订阅号里面，并且订阅号默认按文章展示而不是聊天会话，所以发送消息非常不便。
- 如果你是认证用户，的确可以很方便的在微信聊天列表里发消息。但是这意味着你必须完成企业认证，并且每年都要收费年审。

另外使用上有个小问题，就是微信只能发纯文本、纯图片、存文件，不能组合发。
